webui/react/src/pages/HGICluster.tsx

```diff
@@ -47,6 +63,56 @@ const HGICluster: React.FC = () => {
 
   const slotContainerStates = getSlotContainerStates(agents.data || []);
 
+  const columns = useMemo(() => {
+
+    const newColumns = [ ...defaultColumns ].map(column => {
+
+      const descriptionRender = (_: unknown, record: ResourcePool): React.ReactNode =>
+        <div className={css.descriptionColumn}>{record.description}</div>;
+
+      const chartRender = (_:unknown, record: ResourcePool): React.ReactNode => {
+        const containerStates: ResourceState[] =
+          getSlotContainerStates(agents.data || [], record.name);
+
+        return <SlotAllocationBar
+          className={css.chartColumn}
+          hideHeader
+          resourceStates={containerStates}
+          totalSlots={record.numAgents * record.gpusPerAgent} />;
+
+      };
```

Comment:
**chore:** should be able to move these two renderers outside of the mapping function to avoid having to redefine them over and over.